src/Umbraco.Tests/PropertyEditors/DataValueReferenceFactoryCollectionTests.cs

```diff
@@ -0,0 +1,223 @@
+﻿using Moq;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Umbraco.Core;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Models;
+using Umbraco.Core.Models.Editors;
+using Umbraco.Core.PropertyEditors;
+using Umbraco.Web.PropertyEditors;
+using static Umbraco.Core.Models.Property;
+
+namespace Umbraco.Tests.PropertyEditors
+{
+    [TestFixture]
+    public class DataValueReferenceFactoryCollectionTests
+    {
+        [Test]
+        public void GetAllReferences_All_Variants_With_IDataValueReferenceFactory()
+        {
+            var collection = new DataValueReferenceFactoryCollection(new TestDataValueReferenceFactory().Yield());
+
+            // label does not implement IDataValueReference
```

Comment:
@Shazwazza like your comment says a label is just a string and does not store a reference to anything, but this unit test expects two results/items to be references?!

Reply:
Yes, the point of thi test is to show references resolved from `IDataValueReferenceFactory` which is implemented with `TestDataValueReferenceFactory`, label is used here because it does not implement `IDataValueReference`

Reply:
Thanks did not spot the `TestDataValueReferenceFactory` at the bottom of the test class. All good will merge in :)